src/front-door/azext_front_door/custom.py

```diff
@@ -261,24 +271,40 @@ def update_fd_frontend_endpoints(instance, host_name=None, session_affinity_enab
     return instance
 
 
-def configure_fd_frontend_endpoint_https(cmd, resource_group_name, front_door_name, frontend_endpoint_name,
-                                         disable=None, protocol=None, secret_name=None, secret_version=None,
-                                         certificate_type=None, certificate_source=None, vault=None):
-    if disable:
-        return cf_fd_frontend_endpoints(cmd.cli_ctx, None).disable_https(resource_group_name, front_door_name,
-                                                                         frontend_endpoint_name)
-    # if not being disabled, then must be enabled
+def configure_fd_frontend_endpoint_https_disable(cmd, resource_group_name, front_door_name, item_name):
+    return cf_fd_frontend_endpoints(cmd.cli_ctx, None).disable_https(resource_group_name, front_door_name,
+                                                                     item_name)
+
+
+def configure_fd_frontend_endpoint_https_frontdoor(cmd, resource_group_name, front_door_name, item_name):
+    from azext_front_door.vendored_sdks.models import CustomHttpsConfiguration
+    config = CustomHttpsConfiguration(
+        certificate_source="FrontDoor",
+        protocol_type="ServerNameIndication",
+        vault=None,
+        secret_name=None,
+        secret_version=None,
+        certificate_type="Dedicated"
+    )
+    cf_fd_frontend_endpoints(cmd.cli_ctx, None).enable_https(resource_group_name, front_door_name,
+                                                             item_name, config)
+    return get_fd_frontend_endpoints(cmd, resource_group_name, front_door_name, item_name)
+
+
+def configure_fd_frontend_endpoint_https_keyvault(cmd, resource_group_name, front_door_name, item_name,
```

Comment:
If you plan on supporting other HTTPS configurations, you'll want to consider how your command surface area will expand over time. You could have something like `az network front-door frontend-endpoint configure-https` that would allow you to configure whichever type or HTTPS you want or disable it all with a single command.

Reply:
The problem was with required parameters. HTTPS can be enabled using frontdoor managed certificate or keyvault certificate. Both of them have seperate required parameters. Thats why i had different commands to enable and disable.

Reply:
Group
    az network front-door frontend-endpoint : Manage Front Door frontend endpoints.

Commands:
    create                 : Create a Front Door frontend endpoint.
    delete                 : Delete a Front Door frontend endpoint.
    disable-https          : Disable HTTPS protocol for a custom domain.
    enable-https-frontdoor : Enable HTTPS protocol for a custom domain using Front Door managed
                             certificate.
    enable-https-keyvault  : Enable HTTPS protocol for a custom domain using your own certificate in
                             Azure Key Vault. You need to setup the right permissions for Front Door
                             to access your Key vault. Learn more at
                             https://aka.ms/FrontDoorCustomDomain.
    list                   : List Front Door frontend endpoints.
    show                   : Get the details of a Front Door frontend endpoint.

Reply:
Many commands have this problem and use validators to ensure the user provides the correct set of arguments. The preference in CLI is overloading. The help would look something like:

```
Arguments
    --type             : HTTPS configuration type. Allowed values: FrontDoor, KeyVault, Disabled
KeyVault Arguments
    --keyvault         : Name or ID of the KeyVault containing the certificate.
    --secret-name      : Name of the KeyVault secret.
    --secret-version:  : KeyVault secret version. Omit to use latest.
```

The validator would look something like:
```Python
def process_http_configure_namespace(namespace):
  # keyvault, secret_name, secret_version retrieved from namespace
  keyvault_usage = 'usage error: --type KeyVault --keyvault NAME_OR_ID --secret-name NAME [--secret-version VERSION]'
  if type != 'KeyVault' and any([keyvault, secret_name, secret_version]):
    raise CLIError(keyvault_usage)
  if type == 'KeyVault' and not all([keyvault, secret_name]):
    raise CLIError(keyvault_usage)
```

This would allow you to accomplish all three scenarios with a single command (and support additional ones without creating new commands).